model/generate.go

```diff
@@ -215,41 +215,48 @@ func cacheProjectData(p *Project) projectMaps {
 
 // saveNewBuildsAndTasks saves new builds and tasks to the db.
 func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, cachedProject *projectMaps, v *Version, p *Project, parentPriority int64) error {
-	newTVPairsForExistingVariants := TaskVariantPairs{}
-	newTVPairsForNewVariants := TaskVariantPairs{}
+	// inherit priority from the parent task
+	for i, projBv := range p.BuildVariants {
+		for j := range projBv.Tasks {
+			p.BuildVariants[i].Tasks[j].Priority = parentPriority
+		}
+	}
+
+	newTVPairs := TaskVariantPairs{}
+	for _, bv := range g.BuildVariants {
+		newTVPairs = appendTasks(newTVPairs, bv, p)
+	}
+	newTVPairs.ExecTasks = append(newTVPairs.ExecTasks, IncludePatchDependencies(p, newTVPairs.ExecTasks)...)
+
+	// group into new builds and new tasks for existing builds
 	builds, err := build.Find(build.ByVersion(v.Id).WithFields(build.IdKey, build.BuildVariantKey))
 	if err != nil {
 		return errors.Wrap(err, "problem finding builds for version")
 	}
-	buildSet := map[string]struct{}{}
+	buildSet := make(map[string]bool)
```

Comment:
I'd leave this as a map[string]struct{}{}, which is less ambiguous for set-like cases than maps to bools.

Reply:
done